realm/realm-library/src/objectServer/java/io/realm/SyncUser.java

```diff
@@ -535,7 +535,16 @@ public String toJson() {
      */
     public boolean isValid() {
         Token userToken = getSyncUser().getUserToken();
-        return syncUser.isLoggedIn() && userToken != null && userToken.expiresMs() > System.currentTimeMillis() && SyncManager.getUserStore().isActive(syncUser.getIdentity());
+        return userToken != null && userToken.expiresMs() > System.currentTimeMillis() && !isLoggedOut();
+    }
+
+    /**
+     * Returns whether the user is logged out from the Realm Object Server or not.
+     *
+     * @return  {@code true} if the user was logged out,  {@code false} otherwise.
+     */
+    public boolean isLoggedOut() {
+        return !syncUser.isLoggedIn() || !SyncManager.getUserStore().isActive(syncUser.getIdentity());
```

Comment:
`syncUser.isLoggedIn()` is not necessary here i think? `isActive` will return false when `syncUser.isLoggedIn()` returns false i guess?